inlang/source-code/sdk/src/loadProject.ts

```diff
@@ -80,13 +82,47 @@ export const loadProject = async (args: {
 		// -- settings ------------------------------------------------------------
 
 		const [settings, _setSettings] = createSignal<ProjectSettings>()
-		createEffect(() => {
+		createEffect(async () => {
+			let projectId: string | undefined
+
+			try {
+				projectId = await nodeishFs.readFile(projectPath + "/projectId", {
+					encoding: "utf-8",
+				})
+			} catch (error) {
+				// @ts-ignore
+				if (error.code === "ENOENT") {
+					const repoMeta = await args.repo?.getMeta()
+
+					if (repoMeta && !("error" in repoMeta)) {
+						const idDigest = await crypto.subtle.digest(
+							"SHA-256",
+							new TextEncoder().encode(`${repoMeta.id + projectPath}`)
+						)
+						projectId = [...new Uint8Array(idDigest)]
+							.map((b) => ("00" + b.toString(16)).slice(-2))
+							.join("")
+
```

Comment:
1. Can we turn this code into a dedicated function? 
2. Can you test this function? 
3. Can you place this function outside of the effect? This function doesn't seem to require an effect. This is an initial load problem that does not seem to need reactivity.